app/models/concerns/foreman_remote_execution/host_extensions.rb

```diff
@@ -44,21 +44,11 @@ def execution_status_label(options = {})
       @execution_status_label ||= get_status(HostStatus::ExecutionStatus).to_label(options)
     end
 
+    # rubocop:disable Naming/MemoizedInstanceVariableName
     def host_params_hash
-      params = super
-      keys = remote_execution_ssh_keys
-      source = 'global'
-      if keys.present?
-        value, safe_value = params.fetch('remote_execution_ssh_keys', {}).values_at(:value, :safe_value).map { |v| [v].flatten.compact }
-        params['remote_execution_ssh_keys'] = {:value => value + keys, :safe_value => safe_value + keys, :source => source}
-      end
-      [:remote_execution_ssh_user, :remote_execution_effective_user_method,
-       :remote_execution_connect_by_ip].each do |key|
-        value = Setting[key]
-        params[key.to_s] = {:value => value, :safe_value => value, :source => source} unless params.key?(key.to_s)
-      end
-      params
+      @cached_rex_host_params_hash ||= extend_host_params_hash(super)
```

Comment:
You should probably add this variable to https://github.com/theforeman/foreman/blob/develop/app/models/concerns/host_params.rb#L14 as well